src/universal_transfer_operator/data_providers/database/google/bigquery.py

```diff
@@ -24,15 +72,50 @@ class BigqueryDataProvider(DatabaseDataProvider):
     illegal_column_name_chars_replacement: list[str] = ["_"]
 
     _create_schema_statement: str = "CREATE SCHEMA IF NOT EXISTS {} OPTIONS (location='{}')"
+    NATIVE_PATHS = {
+        FileLocation.GS: "load_gs_file_to_table",
+        FileLocation.S3: "load_s3_file_to_table",
+        FileLocation.LOCAL: "load_local_file_to_table",
+    }
+    NATIVE_AUTODETECT_SCHEMA_CONFIG: Mapping[FileLocation, Mapping[str, list[FileType] | Callable]] = {
+        FileLocation.GS: {
+            "filetype": [FileType.CSV, FileType.NDJSON, FileType.PARQUET],
+            "method": lambda table, file: None,
+        },
+    }
+
+    FILE_PATTERN_BASED_AUTODETECT_SCHEMA_SUPPORTED: set[FileLocation] = {
+        FileLocation.GS,
+        FileLocation.LOCAL,
+    }
+
+    NATIVE_LOAD_EXCEPTIONS: Any = (
+        GoogleNotFound,
+        ClientError,
+        GoogleAPIError,
+        RetryError,
+        InvalidArgument,
+        Unauthorized,
+        Forbidden,
+        Conflict,
+        TooManyRequests,
+        ResourceExhausted,
+        ServerError,
+        Unknown,
+        ServiceUnavailable,
+        InvalidResponse,
+        DatabaseCustomError,
+    )
+    OPTIONS_CLASS = BigqueryOptions
```

Comment:
Where is OPTIONS_CLASS getting used? Is it being referred somewhere outside the module?

Reply:
It is being used here: https://github.com/astronomer/apache-airflow-provider-transfers/blob/502a76a396e543847c2327e5e51861abc4854f12/src/universal_transfer_operator/data_providers/database/google/bigquery.py#L328

Basically, the idea here is to use the `TransferIntegrationOptions` class or its subclass internally, so any dict that is passed to transfer_params should be converted to `TransferIntegrationOptions` or a subclass. More details on this PR: https://github.com/astronomer/apache-airflow-provider-transfers/commit/b489bc72d7dd0b0dfd637c2d50c955ce5055ae68